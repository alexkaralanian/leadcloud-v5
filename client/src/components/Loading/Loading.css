.LoadingContainer {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  margin-top: 2.5rem;
}

.ReactLoading_Icon {
  margin-top: 0px;
  width: 2rem;
  height: 2rem;
}
