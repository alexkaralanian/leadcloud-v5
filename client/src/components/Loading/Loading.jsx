import React from "react";
import { Grid, Row, Col } from "react-bootstrap";
import ReactLoading from "react-loading";
import "./Loading.css";

const Loading = props => (
  <Grid>
    <Row>
      <Col xs={12}>
        <div className="loadingContainer">
          <ReactLoading
            type="spinningBubbles"
            color="#444"
            height={100}
            width={100}
            delay={500}
            className="loadingContainer2"
          />
        </div>
      </Col>
    </Row>
  </Grid>
);

export default Loading;
