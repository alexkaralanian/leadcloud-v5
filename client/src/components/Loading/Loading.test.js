import React from "react";
import { shallow, configure } from "enzyme";
import Loading from "./Loading";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

test("Loading Component renders correctly", () => {
  const component = shallow(<Loading />);
  expect(component).toMatchSnapshot();
});
