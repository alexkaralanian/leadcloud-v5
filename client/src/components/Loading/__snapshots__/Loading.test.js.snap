// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          xs={12}
        >
          <div
            className="loadingContainer"
          >
            <Loading
              className="loadingContainer2"
              color="#444"
              delay="500"
              height="100"
              type="spinningBubbles"
              width="100"
            />
          </div>
        </Col>
      </Row>,
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Col
          bsClass="col"
          componentClass="div"
          xs={12}
        >
          <div
            className="loadingContainer"
          >
            <Loading
              className="loadingContainer2"
              color="#444"
              delay="500"
              height="100"
              type="spinningBubbles"
              width="100"
            />
          </div>
        </Col>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": <div
            className="loadingContainer"
          >
            <Loading
              className="loadingContainer2"
              color="#444"
              delay="500"
              height="100"
              type="spinningBubbles"
              width="100"
            />
          </div>,
          "componentClass": "div",
          "xs": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Loading
              className="loadingContainer2"
              color="#444"
              delay="500"
              height="100"
              type="spinningBubbles"
              width="100"
            />,
            "className": "loadingContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "loadingContainer2",
              "color": "#444",
              "delay": "500",
              "height": "100",
              "type": "spinningBubbles",
              "width": "100",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            xs={12}
          >
            <div
              className="loadingContainer"
            >
              <Loading
                className="loadingContainer2"
                color="#444"
                delay="500"
                height="100"
                type="spinningBubbles"
                width="100"
              />
            </div>
          </Col>
        </Row>,
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <Col
            bsClass="col"
            componentClass="div"
            xs={12}
          >
            <div
              className="loadingContainer"
            >
              <Loading
                className="loadingContainer2"
                color="#444"
                delay="500"
                height="100"
                type="spinningBubbles"
                width="100"
              />
            </div>
          </Col>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": <div
              className="loadingContainer"
            >
              <Loading
                className="loadingContainer2"
                color="#444"
                delay="500"
                height="100"
                type="spinningBubbles"
                width="100"
              />
            </div>,
            "componentClass": "div",
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Loading
                className="loadingContainer2"
                color="#444"
                delay="500"
                height="100"
                type="spinningBubbles"
                width="100"
              />,
              "className": "loadingContainer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "loadingContainer2",
                "color": "#444",
                "delay": "500",
                "height": "100",
                "type": "spinningBubbles",
                "width": "100",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
